kramer/src/main/kotlin/kramer/FetchArtifactCommand.kt

```diff
@@ -138,12 +144,16 @@ internal class FetchArtifactCommand() : CliktCommand(name = "fetch-artifact") {
           "aar" -> {
             linkOrCopy(resolved.pom)
             unzip(resolved.main.localFile, buildFile.parent)
-            write(buildFile, AAR_DOWNLOAD_BUILD_FILE.lines(), CREATE)
+            if (Files.exists(resolved.sources.localFile)) linkOrCopy(resolved.sources)
+            val jars = jarsList(workspace, prefix, Paths.get("classes.jar"), resolved.sources.path)
+            write(buildFile, aarArtifactTemplate(prefix, jars).lines(), CREATE)
           }
           else -> {
             linkOrCopy(resolved.pom)
             linkOrCopy(resolved.main)
-            write(buildFile, fetchArtifactTemplate(prefix, resolved.main.path).lines(), CREATE)
+            if (Files.exists(resolved.sources.localFile)) linkOrCopy(resolved.sources)
+            val jars = jarsList(workspace, prefix, resolved.main.path, resolved.sources.path)
+            write(buildFile, fetchArtifactTemplate(prefix, jars).lines(), CREATE)
```

Comment:
Just for my understanding, 'we're dynamically generating Bazel files that will be later used for downstream compilation?

![image](https://user-images.githubusercontent.com/326857/83183285-68d10700-a0dc-11ea-89c7-e10ecfd416ac.png)

Reply:
Yes. The raw_jvm_import already discovers -sources from within this filegroup, it just wasn't present until now. This lets us not have to make decisions in the `@maven//...` workspace based on the content of this fetch workspace - that would reverse the order of fetching, and force all the .jars to be fetched before we could generate the @maven. This way, only those jars taht are used should be fetched.

(unfortunately .aars do have to be pre-fetched because we need to extract metadata from the AndroidManifest, but I want to avoid propagating that problem to .jar artifacts as well).

Reply:
Actually "we're dynamically generating Bazel files that will be later used for downstream compilation?" - more like later downstream fetching, but yeah - you got the right general principle.